src/Stratis.Bitcoin.Features.SmartContracts/Consensus/SmartContractConsensusFactory.cs

```diff
@@ -1,13 +1,13 @@
 ﻿using NBitcoin;
 
-namespace Stratis.Bitcoin.Features.SmartContracts
+namespace Stratis.Bitcoin.Features.SmartContracts.Consensus
 {
     public class SmartContractConsensusFactory : ConsensusFactory
     {
         /// <inheritdoc />
         public override BlockHeader CreateBlockHeader()
         {
-            return new SmartContractBlockHeader();
+            return new SmartContractPowBlockHeader();
```

Comment:
Could we define a `SmartContractPowConsensusFactory`?

Reply:
Will rename 👍

Reply:
Create a new one :)

Reply:
The current ConsensusFactory is in fact the Pow one ;) Its just named a bit generically but essentially it only produces pow blocks and transactions 👍